client/driver/qemu_test.go

```diff
@@ -148,7 +160,98 @@ func TestQemuDriverUser(t *testing.T) {
 				},
 			},
 		},
-	},
+	}
+
+	ctx := testDriverContexts(t, task)
+	defer ctx.AllocDir.Destroy()
+	d := NewQemuDriver(ctx.DriverCtx)
+
+	dst := ctx.ExecCtx.TaskDir.Dir
+
+	copyFile("./test-resources/qemu/linux-0.2.img", filepath.Join(dst, "linux-0.2.img"), t)
+
+	if _, err := d.Prestart(ctx.ExecCtx, task); err != nil {
+		t.Fatalf("Prestart failed: %v", err)
+	}
+
+	resp, err := d.Start(ctx.ExecCtx, task)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	// The monitor socket will not exist immediately, so we'll wait up to
+	// 5 seconds for it to become available.
+	monitorPath := fmt.Sprintf("%s/linux/%s", ctx.AllocDir.AllocDir, qemuMonitorSocketName)
+	monitorPathExists := false
+	for i := 0; i < 100; i++ {
+		if _, err := os.Stat(monitorPath); !os.IsNotExist(err) {
+			logger.Printf("monitor socket exists at %q\n", monitorPath)
+			monitorPathExists = true
+			break
+		}
+		time.Sleep(200 * time.Millisecond)
+	}
+	if monitorPathExists == false {
+		t.Fatalf("monitor socket did not exist after waiting 20 seconds")
+	}
+
+	// userPid supplied in sendQemuShutdown calls is bogus (it's used only
+	// for log output)
+	if err := sendQemuShutdown(ctx.DriverCtx.logger, "", 0); err == nil {
+		t.Fatalf("sendQemuShutdown should return an error if monitorPath parameter is empty")
+	}
+
+	if err := sendQemuShutdown(ctx.DriverCtx.logger, "/path/that/does/not/exist", 0); err == nil {
+		t.Fatalf("sendQemuShutdown should return an error if file does not exist at monitorPath")
+	}
+
+	if err := sendQemuShutdown(ctx.DriverCtx.logger, monitorPath, 0); err != nil {
+		t.Fatalf("unexpected error from sendQemuShutdown: %s", err)
+	}
+
+	// Clean up
+	defer func() {
```

Comment:
Oh sorry, meant to do this block in a defer right after you check the error from `d.Start`. That way you cleanup even if a fatal error is hit.

Reply:
Double d'oh. Fixed!